File: benchmarks/nim_comparison/strings.nim
proc record(index: int64, active: bool): string =
  result = newStringOfCap(64)
  result.add "customer-"
  result.add $index
  result.add ":active="
  result.add $active
  result.add ":balance="
  result.add $(index * 3)

echo record(0, true)
var total = 0
var index = 0'i64
var active = true
while index < 300_000:
  total += record(index, active).len
  active = not active
  index += 1
echo total

File: benchmarks/php_comparison/strings.php
<?php

declare(strict_types=1);

function record(int $index, bool $active): string
{
    $activeText = $active ? 'true' : 'false';
    return "customer-{$index}:active={$activeText}:balance=" . ($index * 3);
}

echo record(0, true), "\n";

$total = 0;
$active = true;
for ($index = 0; $index < 300_000; $index++) {
    $total += strlen(record($index, $active));
    $active = !$active;
}
echo $total, "\n";
